pkg/controller/provider/container/ovirt/client.go

```diff
@@ -219,3 +221,138 @@ func GetInsecureSkipVerifyFlag(secret *core.Secret) bool {
 
 	return insecureSkipVerify
 }
+
+func TestDownloadOvfStore(secret *core.Secret) (ok bool, err error) {
+
+	// Create the connection to the server:
+	conn, err := ovirtsdk.NewConnectionBuilder().
+		URL(string(secret.Data["url"])).
+		Username(string(secret.Data["user"])).
+		Password(string(secret.Data["password"])).
+		Insecure(true).
+		Compress(true).
+		Timeout(time.Second * 30).
+		Build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	defer conn.Close()
+
+	//Find OVF STORE ID
+	disksService := conn.SystemService().DisksService()
+	diskResponse, err := disksService.List().Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	disks, ok := diskResponse.Disks()
+	var diskIdOVF string
+
+	if ok {
+		//get OVF STORE ID
+		for _, disk := range disks.Slice() {
+			if diskName, ok := disk.Name(); ok {
+				if diskName == "OVF_STORE" {
+					if diskId, ok := disk.Id(); ok {
+						diskIdOVF = diskId
+					}
+					break
+				}
+			}
+		}
+	}
+
+	//in case the OVF STORE is locked wait fo OK status befor stating the image transfer
+	conn.WaitForDisk(diskIdOVF, ovirtsdk.DISKSTATUS_OK, 20*time.Second)
```

Comment:
yes, it assumes that it will always exist, I can add the check for that but is a situation like that can happen at all?

Reply:
it's unlikely - when adding a storage domain, it comes without an OVF store and ovirt-engine periodically updates the OVF store or creates it if it's missing. But otherwise I believe static analysis of this code would raise errors

Reply:
Added a check to see that the OVF id is not empty, and returned false in the case it is.
btw I think that the CI failed due to that, since the fake ovirt doesn't have OVF STORE so the rest of the flow failed and the secret didn't create.